File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
from collections import deque

def bfs(n, computers, i, visit):
    visit[i] = 1
    q = deque([i])
    while q:
        now = q.popleft()
        visit[now] = 1
        for j in range(n):
            if j != now and computers[now][j] == 1:
                if visit[j] == 0:
                    q.append(j)
                

def solution(n, computers):
    answer = 0
    visit = [0]*n
    for i in range(n):
        if visit[i] == 0:
            bfs(n, computers, i, visit)
            answer += 1
    return answer

File: 프로그래머스/lv3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    var visit = new Array(n).fill(false);
    
    function dfs(idx) {
        visit[idx] = true;
        for (let i=0; i<n; i++) {
            if(computers[idx][i] && !visit[i]){
                dfs(i);
            }
        }
    }
    
    for (let k=0; k<n; k++){
        if (!visit[k]){
            dfs(k);
            answer++;
        }
    }
    return answer;
}